docs/Hooks.md

```diff
@@ -4,61 +4,83 @@
 
 Hooks are registered with the `fastify.addHook` method and allow you to listen to specific events in the application or request/response lifecycle. You have to register a hook before the event is triggered otherwise the event is lost.
 
-## Request/Response Hooks
-
 By using the hooks you can interact directly inside the lifecycle of Fastify. There are seven different Hooks that you can use *(in order of execution)*:
-- `'onRequest'`
-- `'preParsing'`
-- `'preValidation'`
-- `'preHandler'`
-- `'preSerialization'`
-- `'onError'`
-- `'onSend'`
-- `'onResponse'`
-
-Example:
+
+- [Request/Reply Hooks](#request/reply-hooks)
+  - [onRequest](#onRequest)
+  - [preParsing](#preParsing)
+  - [preValidation](#preValidation)
+  - [preHandler](#preHandler)
+  - [preSerialization](#preSerialization)
+  - [onError](#onError)
+  - [onSend](#onSend)
+  - [onResponse](#onResponse)
+- [Application Hooks](#application-hooks)
+  - [onClose](#onclose)
+  - [onRoute](#onroute)
+  - [onRegister](#onregister)
+
+**Notice:** the `done` callback is not available when using `async`/`await` or returning a `Promise`. If you do invoke a `done` callback in this situation unexpected behavior may occur, e.g. duplicate invocation of handlers.
+
+## Request/Reply Hooks
+
+[Request](https://github.com/fastify/fastify/blob/master/docs/Request.md) and [Reply](https://github.com/fastify/fastify/blob/master/docs/Reply.md) are the core Fastify objects.<br/>
+`done` is the function to continue with the [lifecycle](https://github.com/fastify/fastify/blob/master/docs/Lifecycle.md).
+
+It is pretty easy to understand where each hook is executed by looking at the [lifecycle page](https://github.com/fastify/fastify/blob/master/docs/Lifecycle.md).<br>
+Hooks are affected by Fastify's encapsulation, and can thus be applied to selected routes. See the [Scopes](#scope) section for more information.
+
+### Manage Errors from a hook
```

Comment:
I think now the structure is near perfection 👏🏼

The `Manage Errors from a hook` and `Respond to a request from a hook` was ok at the bottom as before (after the [onRequest](https://github.com/fastify/fastify/blob/3456c5d2904cdfafb32d983995a3c3b3f3166d0d/docs/Hooks.md#onresponse) ). Sorry for the misunderstanding

Reply:
Done